sympy/sets/fancysets.py

```diff
@@ -1433,6 +1431,11 @@ def _intersect(self, other):
 
     def _union(self, other):
 
+        if other.is_subset(S.Reals):
+
+            # treat a subset of reals as a complex region
+            other = ComplexRegion(other * FiniteSet(0), polar=self.polar)
```

Comment:
Sympy allows negative radius in polar representation, so I thought allowing polar to be true would allow for more simplification (I'll change it to be consistent with mathematical definition).
```python
In [1]: rad, arg = Interval(-5,5), Interval(0, pi)

In [2]: C1 = ComplexRegion(rad * arg, polar=True)

In [3]: C1
Out[3]: {r⋅(ⅈ⋅sin(θ) + cos(θ)) | r, θ ∊ [-5, 5] × [0, π]}
```